project-api/ballerina-projects/src/main/java/io/ballerina/projects/directory/BuildProject.java

```diff
@@ -55,14 +65,56 @@ public static BuildProject loadProject(Path projectPath) {
     private BuildProject(EnvironmentContext environmentContext, Path projectPath) {
         super(environmentContext);
         this.sourceRoot = projectPath;
-        addPackage(projectPath.toString());
+
+        // load Ballerina.toml
+        Path ballerinaTomlPath = this.sourceRoot.resolve(ProjectConstants.BALLERINA_TOML);
+        BallerinaToml ballerinaToml;
+        try {
+            ballerinaToml = BallerinaTomlProcessor.parse(ballerinaTomlPath);
+        } catch (IOException | TomlException e) {
+            throw new RuntimeException(e.getMessage(), e);
+        }
 
         // Set default build options
-//        if (this.context.currentPackage().ballerinaToml().getBuildOptions() != null) {
-//            this.context.setBuildOptions(this.context.currentPackage().ballerinaToml().getBuildOptions());
-//        } else {
-//            this.context.setBuildOptions(new BuildOptions());
-//        }
+        if (ballerinaToml.getBuildOptions() != null) {
+            this.setBuildOptions(ballerinaToml.getBuildOptions());
+        } else {
+            this.setBuildOptions(new BuildOptions());
+        }
+
+        addPackage(projectPath.toString());
+    }
+
+    public BuildOptions getBuildOptions() {
+        return (BuildOptions) super.getBuildOptions();
+    }
+
+    @Nullable
+    public Path modulePath(ModuleId moduleId) {
+        if (currentPackage().moduleIds().contains(moduleId)) {
+            if (currentPackage().getDefaultModule().moduleId() == moduleId) {
+                return sourceRoot;
+            } else {
+                return sourceRoot.resolve(ProjectConstants.MODULES_ROOT).resolve(
+                        currentPackage().module(moduleId).moduleName().moduleNamePart());
+            }
+        }
+        return null;
```

Comment:
What if we use an Optional here, instead of returning null?

Reply:
Fixed in 0d89c8f